File: problems/3033-modify-the-matrix/solution.py
class Solution:
    def modifiedMatrix(self, matrix: List[List[int]]) -> List[List[int]]:
        m = len(matrix)
        n = len(matrix[0])
        answer = [row[:] for row in matrix]
        for j in range(n):
            max_val = -1
            for i in range(m):
                max_val = max(max_val, matrix[i][j])
            for i in range(m):
                if matrix[i][j] == -1:
                    answer[i][j] = max_val
        return answer

File: problems/3033-modify-the-matrix/solution.cpp
class Solution {
public:
    vector<vector<int>> modifiedMatrix(vector<vector<int>>& matrix) {
        int m = matrix.size();
        int n = matrix[0].size();

        for (int j = 0; j < n; ++j) {
            int max_val = -1;
            for (int i = 0; i < m; ++i) {
                max_val = max(max_val, matrix[i][j]);
            }

            for (int i = 0; i < m; ++i) {
                if (matrix[i][j] == -1) {
                    matrix[i][j] = max_val;
                }
            }
        }

        return matrix;
    }
};